scripts/nb-tester/qiskit_docs_notebook_tester/config.py

```diff
@@ -59,143 +103,162 @@ def get_notebook_jobs(args: argparse.Namespace) -> Iterator[NotebookJob]:
+            config_path = Path(args.config_path)
+            config_file = tomllib.loads(config_path.read_text())
         except TypeError as err:
             raise ValueError(
-                f"Couldn't read config from {path}; check it exists and the"
+                f"Couldn't read config from {args.config_path}; check it exists and the"
                 " entries are correct."
             ) from err
 
-    def notebooks_to_execute(self) -> Iterator[Path]:
-        """
-        Yield notebooks to be executed, printing messages for any skipped files.
-        """
-        paths = map(Path, self.args.filenames or self.all_notebooks_to_test)
-        for path in paths:
-            if path.suffix != ".ipynb":
-                print(f"ℹ️ Skipping {path}; file is not `.ipynb` format.")
-                continue
-
-            if matches(path, self.notebooks_exclude):
-                print(
-                    f"ℹ️ Skipping {path}; to run it, edit `notebooks-exclude` in {self.args.config_path}."
-                )
-                continue
-
-            if not self.args.submit_jobs and matches(path, self.notebooks_no_mock):
-                print(
-                    f"ℹ️ Skipping {path} as it doesn't work with mock hardware; use the --submit-jobs flag to run it."
-                )
-                continue
-
-            if self.args.only_submit_jobs and not matches(
-                path, self.all_job_submitting_notebooks
-            ):
-                print(
-                    f"ℹ️ Skipping {path} as it doesn't submit jobs and the --only-submit-jobs flag is set."
-                )
-                continue
-
-            yield path
+        groups = [
+            {"name": key, **value} for key, value in config_file["groups"].items()
+        ]
+        test_strategy = args.test_strategy or config_file["default-strategy"]
+        cell_timeout = (
+            args.cell_timeout or
+            config_file.get("test-strategies", {})
+            .get(test_strategy, {})
+            .get("timeout", None)
+        )
+        return cls(
+            cli_filenames=args.filenames,
+            cell_timeout=cell_timeout,
+            test_strategy=test_strategy,
+            write=args.write,
+            groups=groups,
+        )
 
-    def should_patch(self, path: Path) -> bool:
-        if self.args.submit_jobs:
-            return False
-        return matches(path, self.notebooks_that_submit_jobs)
+    def get_patch_for_group(self, group: dict) -> str | None:
+        patch_config = group["test-strategies"][self.test_strategy]
+
+        patch_name = patch_config.get("patch", None)
+        if patch_name is None:
+            return None
+        if Path(patch_name).exists():
+            return Path(patch_name).read_text().format(**patch_config)
+        if patch_name in BUILT_IN_PATCHES:
+            return BUILT_IN_PATCHES[patch_name].format(**patch_config)
+
+        valid_patch_names = list(BUILT_IN_PATCHES.keys())
```

Comment:
Same comment as above, if we can unify the interface to use files, then we make it easier to add built-in patches